game-core/src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -90,74 +85,37 @@ private void setConditions(final List<RulesAttachment> value) {
     return m_conditions;
   }
 
-  @Override
-  public void clearConditions() {
-    m_conditions.clear();
-  }
-
-  @Override
-  public void resetConditions() {
+  protected void resetConditions() {
     m_conditions = new ArrayList<>();
   }
 
-  @Override
-  @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
-  public void setInvert(final String s) {
-    setInvert(getBool(s));
-  }
-
   @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
   private void setInvert(final boolean s) {
     m_invert = s;
   }
 
-  @Override
-  public boolean getInvert() {
+  private boolean getInvert() {
     return m_invert;
   }
 
-  @Override
-  public void resetInvert() {
-    m_invert = false;
-  }
-
-  @Override
   @GameProperty(xmlProperty = true, gameProperty = true, adds = false)
   public void setConditionType(final String value) throws GameParseException {
-    String s = value;
-    if (s.equalsIgnoreCase("AND")) {
-      s = AND;
-    } else if (s.equalsIgnoreCase("OR")) {
-      s = OR;
-    } else if (s.equalsIgnoreCase("XOR")) {
-      s = XOR;
-    } else {
-      final String[] nums = s.split("-");
-      if (nums.length == 1) {
-        if (Integer.parseInt(nums[0]) < 0) {
-          throw new GameParseException("conditionType must be equal to 'AND' or 'OR' or 'XOR' or 'y' or 'y-z' where Y "
-              + "and Z are valid positive integers and Z is greater than Y" + thisErrorMsg());
-        }
-      } else if (nums.length == 2) {
-        if (Integer.parseInt(nums[0]) < 0 || Integer.parseInt(nums[1]) < 0
-            || !(Integer.parseInt(nums[0]) < Integer.parseInt(nums[1]))) {
-          throw new GameParseException("conditionType must be equal to 'AND' or 'OR' or 'XOR' or 'y' or 'y-z' where Y "
-              + "and Z are valid positive integers and Z is greater than Y" + thisErrorMsg());
-        }
-      } else {
-        throw new GameParseException("conditionType must be equal to 'AND' or 'OR' or 'XOR' or 'y' or 'y-z' where Y "
-            + "and Z are valid positive integers and Z is greater than Y" + thisErrorMsg());
+    final String uppercaseValue = value.toUpperCase();
+    if (uppercaseValue.matches("AND|X?OR|\\d+(?:-\\d+)?")) {
+      final String[] split = uppercaseValue.split("-");
+      if (split.length != 2 || Integer.parseInt(split[1]) > Integer.parseInt(split[0])) {
+        m_conditionType = uppercaseValue;
+        return;
```

Comment:
This is kind of busy.  Some characterization tests on the old implementation would be good to ensure the new implementation produces the same results.

Reply:
Glad that you asked, the only difference seems to be that previously a NumberFormatException was thrown on an invalid input, and now a GameParseException is thrown.
I don't think this makes a difference, because the GameParser should catch all of them.

Reply:
Agreed.